notifier/notify_windows.go

```diff
@@ -1,16 +1,394 @@
+			continue
+		}
+		go handlePipeMessage(conn)
+	}
+}
+
+func handlePipeMessage(conn net.Conn) {
+	data := snoreToastStrToStruct(readWstrConnToStr(conn))
+	log.Debugf("handling PipeMessage: %+v", data)
+
+	if data.action == "timedout" {
+		return
+	}
+
+	notificationsLock.Lock()
+	var n *Notification
+	for _, elem := range notifications {
+		if elem.GUID == data.id {
+			n = elem
+			break
+		}
+	}
+	notificationsLock.Unlock()
+	if n == nil {
+		return
+	}
+	n.Lock()
+	defer n.Unlock()
+
+	switch data.action {
+	case "dismissed": // do nothing
+	case "clicked": // like dismissed: do nothing
+	case "buttonClicked":
+		for _, actions := range n.AvailableActions {
+			if actions.Text == data.button {
+				n.SelectAction(actions.ID)
+				return
+			}
+		}
+
+		log.Warningf("failed to handle button click: button %s is reported to have been clicked but is no registered button. Available options: %+v", data.button, n.AvailableActions)
+	default:
+		log.Debugf("failed to handle SnoreToast-Action %s: not implemented yet", data.action)
+	}
+}
+
+// Snoretoast helper-Functions:
+
+// Verifies that the Argument contians no Semicolon which would be difficult (in some cases impossible) to parse in a Pipe-Response
+func snoreToastVerifyArgumentSyntax(arg string) error {
+	if strings.Contains(arg, ";") {
+		return fmt.Errorf("the SnoreToast-Argument %s would contain a semicolon which would screw up the pipe responses", arg)
+	}
+	return nil
+}
+
+func snoreToastStrToStruct(in string) actionCallback {
+	ret := actionCallback{}
+
+	for _, elem := range strings.Split(in, ";") {
+		if elem == "" {
+			continue
+		}
+
+		elemSplit := strings.SplitN(elem, "=", 2)
+		if len(elemSplit) != 2 {
+			log.Warningf("failing to parse snoretoast-Response %s into Key=Value-Pair; Response: %s", elem, in)
+			continue
+		}
+
+		switch elemSplit[0] {
+		case "action":
+			ret.action = elemSplit[1]
+		case "button":
+			ret.button = elemSplit[1]
+		case "notificationId":
+			ret.id = elemSplit[1]
+		case "version", "pipe": // not needed: do nothing
+		default:
+			log.Infof("failed to parse key %s from SnoreToast-Response into struct: receivedkey is unknown to Portmaster (%s)", elemSplit[0], elem)
+		}
+	}
+
+	if ret.action == "" {
+		log.Errorf("missing attribute action in SnoreToast Response: %s", in)
+	}
+
+	if ret.id == "" {
+		log.Errorf("missing attribute id in SnoreToast Response: %s", in)
+	}
+
+	return ret
+}
+
+func (n *Notification) buildSnoreToastButtonArgument() string {
+	n.Lock()
+	defer n.Unlock()
+
+	temp := make([]string, 0, len(n.AvailableActions))
+	for _, elem := range n.AvailableActions {
+		if err := snoreToastVerifyArgumentSyntax(elem.Text); err != nil {
+			log.Errorf("failed to build SnoreToast Button-Argument: failed to validate Text for %+v: %s", elem, err)
+			continue
+		}
+
+		if elem.Text != "" {
+			temp = append(temp, elem.Text)
+		}
+	}
+	return strings.Join(temp, ";")
+}
+
+func snoreToastRunCmdAsync(cmd *exec.Cmd) {
```

Comment:
I suggest `runSnoreToastCmd` as function name. Makes it easier to read.
Also, as I understand the code, the command is actually executed synchronous. Use `go func()` in go for running something async.